src/Compiler/Service/service.fsi

```diff
@@ -45,7 +47,9 @@ type public FSharpChecker =
         ?enableBackgroundItemKeyStoreAndSemanticClassification: bool *
         ?enablePartialTypeChecking: bool *
         ?enableParallelCheckingWithSignatureFiles: bool *
-        ?parallelReferenceResolution: bool ->
+        ?parallelReferenceResolution: bool *
+        [<Experimental "This parameter is experimental and likely to be removed in the future.">] ?getSource: (string -> ISourceText option) *
+        [<Experimental "This parameter is experimental and likely to be removed in the future.">] ?useChangeNotifications: bool ->
```

Comment:
Are these two attributes coupled in any way?
Or is any combination of the two meaningful?

Reply:
I was thinking about that as well. They make the most sense when used together, but they're not really coupled and can be used separately.

We could say that if `getSource` is given, then it automatically means `useChangeNotifications = true`, but I thought it's better you say that explicitly because it makes it more obvious you have to then also start calling `NotifyFileChanged`.

Reply:
TBH it is not that self-explanatory from the API surface point of view ( = not seeing the source code).

I am thinking if a dedicated DU for meaningful combinations would fit better (and then of course deconstruct into what is already implemented here)

Reply:
I'm not really sure how to improve this with a DU. I only see it either staying this way, or dropping `useChangeNotifications` and just making it `true` whenever `getSource` is given. 

Maybe we could do something like:

```fsharp
type UseFileSystem = Yes | No of (string -> ISourceText option)
```

But you still have to read the doc string to know that you need to call `NotifyFileChanged` if you choose `No`.

Reply:
It could say that in the name of the case, instead of saying "No".

DocumentSource = FileSystem | CallsToNotifyFileChanged of (loadDocument: string -> ISourceTextOption)

Reply:
Well that would make it look like you call that function to do the notification. You can't actually name the parameter (`loadDocument`) there.

So maybe
```fsharp
type DocumentSource = FileSystem | Custom of (string -> ISourceText option)
```

Reply:
I made the change. Still not 100 % convinced, but it probably is better.